### PR TITLE
22754-Exception-while-opening-a-MCSubDirectoryRepository

### PR DESCRIPTION
**Pull request by guillep**
Use FileSystem file references instead of old FileStream

### CODE REVIEW

**Comment by guillep** [comment]
Issue in fogbugz: https://pharo.fogbugz.com/f/cases/resolve/22754/Exception-while-opening-a-MCSubDirectoryRepository

File: src/Monticello/MCSubDirectoryRepository.class.st [hunk 1]
```diff
@@ -75,11 +75,10 @@ MCSubDirectoryRepository >> loadAllFileNames [
...
 { #category : #'i/o' }
 MCSubDirectoryRepository >> readStreamForFileNamed: aString do: aBlock [
-	| file val |
-	file := FileStream readOnlyFileNamed: (self findFullNameForReading: aString).
-	val := aBlock value: file.
-	file close.
-	^ val
+
+	| fileReference |
+	fileReference := (self findFullNameForReading: aString) reference.
```
Comment (svenvc) [inline src/Monticello/MCSubDirectoryRepository.class.st:80]: sure about that #reference send ?

**Comment by svenvc** [comment]
sure about that #reference send ? 

the binary streams are needed for MC zips I guess.

File: src/Monticello/MCSubDirectoryRepository.class.st (hunk 1 above)
Comment (guillep) [reply src/Monticello/MCSubDirectoryRepository.class.st:80]: Yes, `findFullNameForReading:` returns a `FileSystemDirectoryEntry`.

**Comment by guillep** [comment]
> the binary streams are needed for MC zips I guess.

Exactly

**Comment by svenvc** [comment]
ok, thanks

**Review by pavel-krivanek — APPROVED** [review]

**Pull request merged**